lib/cli/analytics.js

```diff
@@ -46,61 +53,50 @@ var AnalyticsOffMessage  = '\nAnalytics is off. \nIf you would like to turn anal
 var AnalyticsOnMessage  = '\nAnalytics is on! Nice, you are helping us improve PhoneGap tooling.\n' +
                          'If you would like to turn analytics off, simply run `phonegap analytics off`\n';
 
+/*
+    Helper function to determine if cli is a production/dev release
+*/
 
-module.exports = function analytics(argv, callback) {
-    var params = argv._.slice(1);
-
-    if (params.length > 0) {
-        switch(params[0]) {
-            case "on" :
-                // only track it if the value is changing
-                if (insight.optOut !== false) {
-                    // must change it before we use it
-                    insight.optOut = false;
-                    insight.trackEvent({"category":category,
-                                        "action":"set analytics",
-                                        "label":"opt in"});
-                }
-                break;
-            case "off" :
-                // track it only if it changes
-                if (insight.optOut !== true) {
-                    // must use it before we change it
-                    insight.trackEvent({"category":category,
-                                        "action":"set analytics",
-                                        "label":"opt out"});
-                    insight.optOut = true;
-                }
-                break;
-            case "reset" :
-                // this is a secret api, mainly just for testing the prompt messages
-                console.log("resetting it .. ");
-                insight.optOut = undefined;
-                // intentional fallthrough!
-            default:
-                // unrecognized param
-                insight.trackEvent({"category":category,
-                                    "action":"set analytics",
-                                    "label":"unrecognized param"});
-                break;
-        }
+function getDebugFlag() {
+    if(/cordova/.test(category)) {
```

Comment:
a) why not place the test inline where it used, since it is a one liner
b) if it must be a function, why not pass the category into it, so it is not side effecty
c) what do we expect valid categories to be ( this is where I was confused ) 
 - the rest is just me being picky ...